### PR TITLE
Add check to prevent log download occurring when the battery level is updated

### PR DESCRIPTION
**Pull request by camroots**
**Related issue**
https://github.com/openmsupply/msupply-cold-chain/issues/229

**PR changes**
This PR covers the following changes
- Updates the npm packages to resolve build issues.
- Updates the msupply-ble-service files via `yalc`.
- Adds the selector `isSensorUpdating` to prevent a download occurring if the battery level is currently being read.

**Details**
Based on the current implementation, when logs are downloaded from a sensor, the `connectAndDiscoverServices` method (from the `msupply-ble-service`) is called first. This method cancels the current connection if one exists and then reconnects. 
The `connectAndDiscoverServices` method is called when downloading logs and when retrieving the battery level. As these two operations happen concurrently, it is possible for them to overlap. This is most likely to occur when the app first starts logging.
There exists the selector `isSensorDownloading` which is used to prevent the battery being updated if a log download is occurring. Previously, there was no check for the opposite case, where the battery is updating and logging begins. The `isSensorUpdating` has been added and is used to prevent a download from occurring.

After testing a debug version I no longer see errors such as:
image

### CODE REVIEW

**Review by jmbrunskill — APPROVED** [review]
Looks like some good debugging, Hopefully the sensorUpdating thing helps!

**Comment by andreievg** [comment]
@camroots nice, this definitely looks like an improvement, I would assume it's attempting to deal with this issue: https://github.com/openmsupply/msupply-cold-chain/issues/229 ?

If so I was wondering, how it would achieve that task ? Wouldn't app try to re-sync the logs in the next schedule ? Or does this code allow deletion of logs even if they haven't been synced ? (if the later is the issue, and this PR fixes it, we should be able to replicate it by causing the connection error pragmatically ?). 

Also does should there be a general flag for any operations (for example to not allow battery check connection when general log sync is in progress ?)

Lastly to check that the log disappears, is there a schedule configuration we can use (like very fast battery check interval ?)

**Comment by andreievg** [comment]
I made an issue base [Separate Thought](https://github.com/openmsupply/msupply-cold-chain/issues/233), it's good to flag it in a more permanent place. Thanks for that thought

File: .yalc/msupply-ble-service/lib/commonjs/Bluetooth/BleService.js [hunk 1]
```diff
@@ -38,6 +38,9 @@ const dummyLogger = {
     /*do nothing*/
   }
 };
+const RETRY_DELAY = 1000;
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply .yalc/msupply-ble-service/lib/commonjs/Bluetooth/BleService.js:41]: Can you import and use `MILLISECONDS.ONE_SECOND` instead please. In fact, given that all usages of `sleep` are for `1000` you could just have that in the fn definition 🤷 
e.g. `await retry_delay()`

File: android/.project [hunk 2]
```diff
@@ -16,12 +16,12 @@
 	</natures>
 	<filteredResources>
 		<filter>
-			<id>1622026265925</id>
+			<id>1683102094740</id>
```
Comment (mark-prins) [inline android/.project:19]: 🤔 🤷

**Review by mark-prins — COMMENTED** [review]
👍 Nice! can't have too much logging!
I'd added the logger and a lot of logging when finding the previous log download issue - good to see that it's useful.

**Comment by andreievg** [comment]
@camroots, thank you for description in the issue, I think solution proposed in follow up comment is the way to resolve it: https://github.com/openmsupply/msupply-cold-chain/issues/229#issuecomment-1556166880. Change in this PR is also helpful, but I am not sure what the full effect of it is, and not sure I have the time to do a full review, so up to existing reviews if it ok to merge or not.

**Comment by adamdewey** [comment]
@mark-prins - do you reckon this is ok to merge without Andrei's eagle eye?

Am hoping to get something out to the client asap as they're pulling their hair out...

**Pull request merged**